x-pack/agent/docs/agent_configuration_example.yml

```diff
@@ -0,0 +1,518 @@
+         - metricset: memory
+           dataset: system.memory
+         - metricset: diskio
+           dataset: system.diskio
+         - metricset: load
+           dataset: system.load
+         - metricset: memory
+           dataset: system.memory
+         - metricset: process
+           dataset: system.process
+           processes: ["firefox*"]
+         - metricset: process_summary
+           dataset: system.process_summary
+         - metricset: uptime
+           dataset: system.uptime
+         - metricset: socket_summary
+           dataset: system.socket_summary
+         - metricset: filesystem
+           dataset: system.filesystem
+         - metricset: raid
+           dataset: system.raid
+         - metricset: socket
+           dataset: system.socket
+         - metricset: service
+           dataset: system.service
+         - metricset: fsstat
+           dataset: system.fsstat
+         - metricset: foo
+           dataset: system.foo
+
+
+  #################################################################################################
+  # Elasticsearch package example
+  - id?: my-endpoint
+    title: Collect Elasticsearch information
+    package:
+      name: epm/elsticsearch
+      version: 1.7.0
+    inputs:
+        - type: log
+          streams:
+          - id?: {id}
+            enabled?: true # default to true
+            dataset: elasticsearch.audit
+            paths: [/var/log/elasticsearch/*_access.log, /var/log/elasticsearch/*_audit.log]
+          - id?: {id}
+            enabled?: true
+            dataset: elasticsearch.deprecation
+            paths: [/var/log/elasticsearch/*_deprecation.log]
+          - id?: {id}
+            enabled?: true
+            dataset: elasticsearch.gc
+            paths: [/var/log/elasticsearch/*_gc.log, /var/log/elasticsearch/*_gc.log.[0-9]*]
+          - id?: {id}
+            enabled?: true
+            dataset: elasticsearch.server
+            paths: [/var/log/elasticsearch/*.log]
+          - id?: {id}
+            enabled?: true
+            dataset: elasticsearch.slowlog
+            paths: [/var/log/elasticsearch/*_index_search_slowlog.log, /var/log/elasticsearch/*_index_indexing_slowlog.log]
+        - type: metrics/elasticsearch
+          hosts: ["http://localhost:9200"]
+          # api_key: "xxxx"
+          xpack.enabled: true
+          streams:
+          - id?: {id}
+            metricset: ccr
+            dataset: elasticseach.ccr
+          - id?: {id}
+            metricset: cluster_stats
+            dataset: elasticseach.cluster_stats
+          - id?: {id}
+            metricset: enrich
+            dataset: elasticseach.enrich
+          - id?: {id}
+            metricset: index
+            dataset: elasticseach.index
+          - id?: {id}
+            metricset: index_recovery
+            dataset: elasticseach.index_recovery
+          - id?: {id}
+            metricset: index_recovery
```

Comment:
Lets add the index_recovery.active_only here to make it clear, there are also stream specific configs: https://github.com/elastic/beats/blob/master/metricbeat/module/elasticsearch/_meta/config.reference.yml#L17

My goal is to have some complete examples.

Reply:
I have added a few `ssl.*` and the username/password.